service/debugger/debugger.go

```diff
@@ -1475,6 +1487,158 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
+
+func filter(fname string) bool {
+	idx := strings.Index(fname, "runtime.")
+	idx2 := strings.Index(fname, "runtime.defer")
+	idx3 := strings.Index(fname, "runtime.gorecover")
+	idx4 := strings.Index(fname, "runtime.gopanic") 
+	// Avoid runtime/internal functions to figure in the trace function list as they dont have a stack associated
+	idx5 := strings.Index(fname, "runtime/internal")
+	if (idx != -1 || idx5 != -1) && idx2 == -1 && idx3 == -1 && idx4 == -1 {
+		// Except certain special functions do not traverse runtime.* functions
+		return false
+	}
+	return true
+}
+
+func dequeue(queue *list.List) TraceFuncptr {
+	element := queue.Front()
+	queue.Remove(element)
+	data := element.Value
+	F, ok := data.(TraceFuncptr)
+	if !ok {
+		fmt.Printf("Type error\n")
+		os.Exit(1)
+	}
+	return F
+}
+
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
+
+	var queue *list.List
+	queue = list.New()
+	queue.Init()
+
+	funcs := []string{}
+
+	rootnode := &TraceFunc{Func: new(proc.Function), Depth: depth, children: []*proc.Function{}, visited: false, dis: false}
+	rootnode.Func = f
+
+	TraceMap[f.Name] = rootnode
+	queue.PushBack(TraceMap[f.Name])
+	for queue.Len() > 0 {
+		parent := dequeue(queue)
+		if parent.Depth > FollowCalls {
+			continue
+		}
+		if !parent.visited {
+			// Guard to avoid adding duplicates to funcs
+			funcs = append(funcs, parent.Func.Name)
+			parent.visited = true
+		}
+		if parent.Depth+1 > FollowCalls {
+			// Skip disassembling and computing children if we are going to cross FollowCalls
+			continue
+		}
+		deferdepth := 0
+		var deferparent TraceFuncptr = nil
+		if !parent.dis {
+			f := parent.Func
+			text, err := proc.Disassemble(t.Memory(), nil, t.Breakpoints(), t.BinInfo(), f.Entry, f.End)
+			if err != nil {
+				return nil, fmt.Errorf("disassemble failed with error %w", err)
+			}
+			for _, instr := range text {
+				if instr.IsCall() && instr.DestLoc != nil && instr.DestLoc.Fn != nil {
+					cf := instr.DestLoc.Fn
+					if !filter(cf.Name) {
+						continue
+					}
+					childnode := TraceMap[cf.Name]
+					if childnode == nil {
+						childnode = &TraceFunc{Func: new(proc.Function), Depth: parent.Depth + 1, children: []*proc.Function{}, visited: false, dis: false}
+						childnode.Func = cf
+						TraceMap[cf.Name] = childnode
+						parent.children = append(parent.children, cf)
+					} else {
+						if childnode.Depth > parent.Depth+1 {
+							// can go even further deep
+							childnode.Depth = parent.Depth + 1
+							parent.children = append(parent.children, cf)
+						}
+					}
+
+					if strings.Contains(cf.Name, "runtime.deferreturn") {
+						if deferparent == nil && childnode.Depth+1 <= FollowCalls {
+							// keep track of defer function inorder to trace defer calls
+							deferdepth = childnode.Depth
+							deferparent = parent
+						}
+					}
+
+				}
+			}
+			parent.dis = true
+		}
+		for _, cf := range parent.children {
```

Comment:
(as noted in an earlier comment) I think this loop is unnecessary and we can directly enqueue in the above loop.